File: 백준/Gold/24492. Cow Frisbee/Cow Frisbee.cc
//Cow Frisbee
#include <iostream>
#include <stack>
#include <vector>

#define ll long long
using namespace std;

int n;
int a,idx;
ll res;

stack<pair<int,int>> st;

int main(){
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);

    cin >> n;
    for (int i=0; i<n; i++) {
        cin >> a;
        while (!st.empty() && st.top().first < a) {
            idx = st.top().second;
            st.pop();
            res += i-idx+1;
        }

        if (!st.empty()) {
            res += i-st.top().second+1;
        }

        st.push({a,i});
    }

    cout << res;

    return 0;
}

File: 백준/Gold/24492. Cow Frisbee/Cow Frisbee.py
#Cow Frisbee
n = int(input())
d = list(map(int,input().split()))

res = 0
stack = []
for i in range(n):
    while stack and stack[-1][0] < d[i]:
        idx = stack.pop()[1]
        res += i-idx+1

    if stack:
        res += i-stack[-1][1]+1

    stack.append((d[i],i))

print(res)